docs/system-overview.rst

```diff
@@ -0,0 +1,84 @@
+Proctoring System Overview
+===========================
+This document outlines the components involved in the edX proctoring system. It should
+serve as a guide to orient developers on what components exist, how they interact, and 
+where to find the source code in our platform.
+
+.. contents::
+
+System Components
+------------------
+
+.. image:: images/components.png
+
+Proctored Exam Views
+^^^^^^^^^^^^^^^^^^^^
+
+Interstitial views within a section that are shown to in place of the actual
+exam content. They are used to walk a leaner through setup steps
+and display the state of the current attempt when the exam is not in progress.
+
+The LMS will call into edx-proctoring to load the relevent template when rendering the
+student view for an exam section that is proctored.
+
+Notable Code:
+
+- `LMS render student_view() <https://github.com/edx/edx-platform/blob/a7dff8c21ee794e90bdc0f22876334a7843a032d/common/lib/xmodule/xmodule/seq_module.py#L274>`_
+- `edx-proctoring template logic <https://github.com/edx/edx-proctoring/blob/78976d93ab6ca5206f259dc420d2f45818fe636c/edx_proctoring/api.py#L1912>`_
+- `edx-proctoring interstitial templates <https://github.com/edx/edx-proctoring/tree/master/edx_proctoring/templates>`_
+
+JavaScript Message API and Worker
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A set of functions called by edX courseware to emit messages based on changing
+exam states. These messages may be handled by a JS worker specific to the proctoring provider. 
+That worker is included as part of the the provider's python plugin and will 
+interface directly with the proctoring software running on the learner's machine. Exam
+state will transition forward only after these messages have been successfully handled.
+However, if a provider does not have a worker interface configured there will be no direct
+communication between edX courseware and the proctoring client.
+
+Message Interface: `exam_action_handler.js <https://github.com/edx/edx-proctoring/blob/master/edx_proctoring/static/proctoring/js/exam_action_handler.js>`_
+
+Example worker: `proctortrack_custom.js <https://github.com/joshivj/edx-proctoring-proctortrack/blob/master/edx_proctoring_proctortrack/static/proctortrack_custom.js>`_
+
+edx-proctoring
+^^^^^^^^^^^^^^
+Python plugin that handles the bulk of edX's proctoring logic. It hosts the models for proctored
+exam configuration and learner attempts.  It exposes a REST and Python interface to manage them.
+edx-proctoring is also responsible for calling out to the provider's backend (through a plugin) to keep
+exam configuration and learner attempts in-sync between the two systems.
+
+https://github.com/edx/edx-proctoring/
```

Comment:
I don't feel like this link is useful. Maybe link the code to https://github.com/edx/edx-proctoring/blob/master/edx_proctoring/views.py#L288 is better?